addons/website/models/website_page.py

```diff
@@ -229,6 +237,94 @@ def _set_cache_response(self, cache_key, response):
         """ Put in cache the given response. """
         self.pool._Registry__cache[('website.page', _cached_response, self.id, cache_key)] = response
 
+    @api.model
+    def _search_get_detail(self, website, order, options):
+        with_description = options['displayDescription']
+        requires_sudo = False
+        domain = [website.website_domain()]
+        if not self.env.user.has_group('website.group_website_designer'):
+            domain.append([('website_published', '=', True)])
+            requires_sudo = True
+
+        search_fields = ['name', 'url']
+        fetch_fields = ['id', 'name', 'url']
+        mapping = {
+            'name': {'name': 'name', 'type': 'text', 'match': True},
+            'website_url': {'name': 'url', 'type': 'text'},
+        }
+        if with_description:
+            search_fields.append('arch_db')
+            fetch_fields.append('arch')
+            mapping['description'] = {'name': 'arch', 'type': 'text', 'html': True, 'match': True}
+        return {
+            'model': 'website.page',
+            'base_domain': domain,
+            'requires_sudo': requires_sudo,
+            'search_fields': search_fields,
+            'fetch_fields': fetch_fields,
+            'mapping': mapping,
+            'icon': 'fa-file-o',
+        }
+
+    @api.model
+    def _search_fetch(self, search_detail, search, limit, order):
+        with_description = 'description' in search_detail['mapping']
+        results, count = super()._search_fetch(search_detail, search, limit, order)
+        if with_description and search:
+            # Perform search in translations
+            # TODO Remove when domains will support xml_translate fields
+            query = sql.SQL("""
+                SELECT {table}.{id}
+                FROM {table}
+                LEFT JOIN ir_ui_view v ON {table}.{view_id} = v.{id}
+                LEFT JOIN ir_translation t ON v.{id} = t.{res_id}
+                WHERE t.lang = {lang}
+                AND t.name = ANY({names})
+                AND t.type = 'model_terms'
+                AND t.value ilike {search}
+                LIMIT {limit}
+            """).format(
+                table=sql.Identifier(self._table),
+                id=sql.Identifier('id'),
+                view_id=sql.Identifier('view_id'),
+                res_id=sql.Identifier('res_id'),
+                lang=sql.Placeholder('lang'),
+                names=sql.Placeholder('names'),
+                search=sql.Placeholder('search'),
+                limit=sql.Placeholder('limit'),
+            )
+            self.env.cr.execute(query, {
+                'lang': self.env.lang,
+                'names': ['ir.ui.view,arch_db', 'ir.ui.view,name'],
+                'search': '%%%s%%' % search,
```

Comment:
There is one oddity here: `search` comes from `_search_exact` (which incidentally might be ill-named?) which forwards it from `_search_with_fuzzy` which gets it from its caller, possibly processed through `_search_find_fuzzy_term`.

The caller is generally the controllers, so this is an arbitrary `ilike` pattern under foreign control, is that intended? `_search_build_domain` explicitly escapes `search_term` so it seems like a divergence.

An other possibly odd bit (of both) is `%xxx%` will match `xxx` in the middle of a word, but I'm not sure there is a good way to fix it, and here the two versions are coherent.

Reply:
Thank you, I added the missing escaping and verified that it allows fetching less pages to deliver the correct result.

Regarding the `_search_exact` method name, it is related to the way the caller sees it: within that method the search term is not approximated anymore, it is only used "as provided". (As opposed to "search for something close enough according to fuzzy logic"). Maybe `_search_exact_term_only` would be more explicit, but the naming was intentional.

Regarding the matching in the middle of words, there was a discussion with the PO about it, and it actually became a requirement, so that you can look for "phone" in a music instruments shop and get "saxophone", "vibraphone" and "xylophone" as results. [Example recovered from Discord]

Reply:
OK.

Reply:
@xmo-odoo  Ok, but my build is still red security-wise because of the `new Markup(...)` on the client-side. Can it be whitelisted ?